### PR TITLE
[VeOmni] Standalone dense P-EAGLE drafter training on the VeOmni backend

### PR DESCRIPTION
**Pull request by 0z5a**
## What this adds

Standalone VeOmni execution for dense P-EAGLE drafters, so the drafter side of the roadmap's
VeOmni item can run on the VeOmni backend instead of only FSDP2.

- `verl_speco/trainer/veomni_drafter.py` — VeOmni drafter training entry with explicit optimizer-step
  boundaries and checkpoint save/resume.
- `verl_speco/backends/peagle_trainer_backend.py`, `verl_speco/trainer/base_trainer.py`,
  `verl_speco/config/speco_base.yaml` — backend selection and configuration for the VeOmni path.
- Regression tests for backend selection and the P-EAGLE contract.

## Why it is not duplicating existing work

Searched `gh pr list --repo verl-project/verl-SpeCo --state open --search "veomni"` and the issue
list: the only VeOmni PR is #54 (DSpark co-training with VeOmni actors, merged), which covers the
**actor** side. This branch is the **drafter** side, the scope the roadmap owner asked for in #7.
No open PR covers standalone dense P-EAGLE drafter training on VeOmni.

## Validation

Runtime: Python 3.12, PyTorch 2.13.0, vLLM 0.29.0, two ranks.

| Check | Result |
|---|---|
| Focused regression tests (`tests/integration/test_veomni_drafter_selection.py`) | 5 passed |
| Two-rank optimizer-step parity against the FSDP2 path | passed |
| Six-step standalone training with saves at steps 3 and 6 | passed |
| Separate resume run from the step-3 checkpoint | passed |
| TP1/TP2 serving from the trained checkpoint (eager and graph) | passed |

Raw timing and correctness logs for the lifecycle runs are in this branch.

```bash
python -m pytest tests/integration/test_veomni_drafter_selection.py -q
```

## Scope limits

Dense P-EAGLE only. No online actor/rollout integration claim, and the standalone totals are single
observations on a shared machine, not a throughput measurement.

## AI assistance

AI assistance was used for the implementation, tests and this description. The submitter reviewed
the change and ran the commands above.

### CODE REVIEW

**Comment by 0z5a** [comment]
Self-review of the changed lines.

- `veomni_drafter.py`: the training entry drives optimizer steps through the backend's own step boundary; no optimizer state is shared with the FSDP2 path, and checkpoint save/resume goes through the existing checkpoint helpers rather than a parallel writer.
- Backend selection: the VeOmni path is opt-in through config; with the flag off the FSDP2 path is byte-identical to before this branch (`git diff origin/main -- verl_speco/trainer/base_trainer.py` shows only the selection call site).
- No broad exception handling, no `Any` to bypass typing, and no `getattr` reads of fields that already exist on the dataclasses involved.
- Tests: `tests/integration/test_veomni_drafter_selection.py` covers flag on/off selection and the P-EAGLE backend contract; I re-ran them on a clean RTX 5090 host and got 5 passed.
- Not claimed: online actor/rollout integration, packed/variable-length edge cases beyond the recorded fixture, and any throughput conclusion from the single shared-host comparison.

Known environment note for reviewers: this host has no `flash_attn`, so the recorded runs used a pure-PyTorch padding shim; that shim is not part of this branch.
